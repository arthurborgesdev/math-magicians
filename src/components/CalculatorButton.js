import React from 'react';
import PropTypes from 'prop-types';

class CalculatorButton extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleClick = this.handleClick.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
  }

  handleClick() {
    const { text, onClick, calcObj } = this.props;
    onClick(calcObj, text);
  }

  handleKeyDown() {
    const { text, onClick, calcObj } = this.props;
    onClick(calcObj, text);
  }

  render() {
    const { text } = this.props;
    return (
      <button
        type="button"
        tabIndex={0}
        className="keyboard-button"
        onClick={this.handleClick}
        onKeyDown={this.handleKeyDown}
      >
        {text}
      </button>
    );
  }
}

CalculatorButton.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  calcObj: PropTypes.objectOf([
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.object,
    ]),
  ]).isRequired,
};

// PropTypes.checkPropTypes(CalculatorButton, props, 'prop', 'CalculatorButton');

export default CalculatorButton;
