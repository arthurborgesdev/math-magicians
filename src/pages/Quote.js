import React from 'react';

const Quote = () => (
  <div className="quote-container">
    <h4>
      Mathematics is not about numbers, equations, computations, or
      algorithms: it is about understanding. - William Paul Thurston
    </h4>
  </div>
);

export default Quote;
