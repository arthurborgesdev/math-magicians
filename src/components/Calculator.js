import React, { useState } from 'react';
import CalculatorButton from './CalculatorButton';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculatorObj, setCalculatorObj] = useState({});

  const handleCalculate = (calcObj, text) => {
    setCalculatorObj(calculate(calcObj, text));
  };

  let display = 0;
  if (Object.keys(calculatorObj).length === 0 || (calculatorObj.total === null
    && calculatorObj.next === null
    && calculatorObj.operation === null)) {
    display = 0;
  } else {
    display = calculatorObj.next !== null ? calculatorObj.next : calculatorObj.total;
  }

  return (
    <div className="calculator-component">
      <div className="calculator-display">
        { display }
      </div>
      <div className="calculator-keyboard">
        <div className="calculator-digits">
          <CalculatorButton text="AC" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="+/-" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="%" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="7" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="8" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="9" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="4" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="5" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="6" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="1" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="2" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="3" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="0" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="." onClick={handleCalculate} calcObj={calculatorObj} />
        </div>
        <div className="calculator-operators">
          <CalculatorButton text="รท" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="x" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="-" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="+" onClick={handleCalculate} calcObj={calculatorObj} />
          <CalculatorButton text="=" onClick={handleCalculate} calcObj={calculatorObj} />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
