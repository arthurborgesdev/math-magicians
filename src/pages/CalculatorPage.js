import React from 'react';
import Calculator from '../components/Calculator';

const CalculatorPage = () => {
  return (
    <>
      <h1>Let's do some math!</h1>
      <Calculator />
    </>
  )
}

export default CalculatorPage; 