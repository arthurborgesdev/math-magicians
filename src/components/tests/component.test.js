import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Calculator from '../Calculator';
import NavBar from '../NavBar';
import CalculatorButton from '../CalculatorButton';
import CalculatorPage from '../../pages/CalculatorPage';
import Home from '../../pages/Home';
import Quote from '../../pages/Quote';

describe('Testing all components rendering', () => {
  it('Calculator renders correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('NavBar renders correctly', () => {
    const tree = renderer
      .create(<Router><NavBar /></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('CalculatorButton renders correctly', () => {
    let text = 'test';
    let onClick = jest.fn(() => {});
    let calcObj = { string: 'test' };
    const tree = renderer
      .create(<CalculatorButton text={text} onClick={onClick} calcObj={calcObj} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('CalculatorPage renders correctly', () => {
    const tree = renderer
      .create(<CalculatorPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Home page renders correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Quote page renders correctly', () => {
    const tree = renderer
      .create(<Quote />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
