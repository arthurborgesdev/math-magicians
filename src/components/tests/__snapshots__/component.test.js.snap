// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing all components rendering Calculator renders correctly 1`] = `
<div
  className="calculator-component"
>
  <div
    className="calculator-display"
  >
    0
  </div>
  <div
    className="calculator-keyboard"
  >
    <div
      className="calculator-digits"
    >
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="calculator-operators"
    >
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="keyboard-button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing all components rendering CalculatorButton renders correctly 1`] = `
<button
  className="keyboard-button"
  onClick={[Function]}
  type="button"
/>
`;

exports[`Testing all components rendering CalculatorPage renders correctly 1`] = `
<div
  className="calculator-container"
>
  <h2>
    Let's do some math!
  </h2>
  <div
    className="calculator-component"
  >
    <div
      className="calculator-display"
    >
      0
    </div>
    <div
      className="calculator-keyboard"
    >
      <div
        className="calculator-digits"
      >
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </div>
      <div
        className="calculator-operators"
      >
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="keyboard-button"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing all components rendering Home page renders correctly 1`] = `
Array [
  <h2>
    Welcome to our page!
  </h2>,
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>,
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>,
]
`;

exports[`Testing all components rendering NavBar renders correctly 1`] = `
<div
  className="navbar-component"
>
  <h1>
    Math Magicians
  </h1>
  <nav
    className="navbar"
  >
    <ul
      className="navbar-list"
    >
      <li
        className="navbar-list-items"
      >
        <a
          aria-current="page"
          className="active-link"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <span
          className="pipe-divisor"
        >
           | 
        </span>
      </li>
      <li
        className="navbar-list-items"
      >
        <a
          aria-current={null}
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
        <span
          className="pipe-divisor"
        >
           | 
        </span>
      </li>
      <li
        className="navbar-list-items"
      >
        <a
          aria-current={null}
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
        <span
          className="pipe-divisor"
        >
           | 
        </span>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Testing all components rendering Quote page renders correctly 1`] = `
<div
  className="quote-container"
>
  <h3>
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston
  </h3>
</div>
`;
